components/brave_new_tab_ui/components/default/cryptoDotCom/index.tsx

```diff
@@ -451,35 +421,690 @@ class CryptoDotCom extends React.PureComponent<Props, State> {
+          }
+        </WidgetWrapper>
+      </ThemeProvider>
     )
   }
 }
 
 export const CryptoDotComWidget = createWidget(CryptoDotCom)
+
+// Supporting Components
+
+function AssetDeposit ({
+  assetAddress,
+  assetQR,
+  base,
+  handleBackClick
+}: any) {
+  const [showQR, setShowQR] = React.useState(false)
+
+  return showQR
+  ? (
+    <Box isFlex={true} column={true} $p={10}>
+      <img src={assetQR} />
+      <ActionButton onClick={() => setShowQR(false)} $mt={10} small={true} light={true} isFullWidth={false}>
+        Done
+      </ActionButton>
+    </Box>
+  ) : (
+    <Box $p={0}>
+      <FlexItem
+        hasPadding={true}
+        isFlex={true}
+        isFullWidth={true}
+        hasBorder={true}
+      >
+        <FlexItem>
+          <BackArrow onClick={handleBackClick}>
+            <CaratLeftIcon />
+          </BackArrow>
+        </FlexItem>
+        <FlexItem $pr={5}>
+          {renderIconAsset(base.toLowerCase())}
+        </FlexItem>
+        <FlexItem flex={1}>
+          <Text>{base}</Text>
+          <Text small={true} textColor='light'>
+            {currencyNames[base]}
+          </Text>
+        </FlexItem>
+        <FlexItem $pl={5}>
+          <PlainButton onClick={() => setShowQR(true)}>
+            <img width={25} src={QRIcon} />
+          </PlainButton>
+        </FlexItem>
+      </FlexItem>
+      <FlexItem
+        $p='0.5em'
+        isFullWidth={true}
+      >
+        <Text $fontSize={13} weight={600}>{base} Address</Text>
+        <Text $fontSize={13} breakWord={true}>{assetAddress}</Text>
+        <ActionButton onClick={() => copyToClipboard(assetAddress)} $mt={5} $mb={15} small={true} light={true} isFullWidth={false}>
+          {getLocale('cryptoDotComWidgetCopyAddress')}
+        </ActionButton>
+        <Text $fontSize={13} weight={600}>{getLocale('cryptoDotComWidgetSendCaveatHeading', { currency: base })}</Text>
+        <Text $fontSize={13}>{getLocale('cryptoDotComWidgetSendCaveatBody', { currency: base })}</Text>
+      </FlexItem>
+    </Box>
+  )
+}
+
+function AssetTrade ({
+  base,
+  quote,
+  availableBalanceBase,
+  availableBalanceQuote,
+  handleBackClick,
+  tickerPrices,
+  priceDecimals,
+  quantityDecimals,
+}: Record<string, any>) {
+
+  enum TradeModes {
+    BUY = 'buy',
+    SELL = 'sell'
+  }
+
+  const tradeModeLocaleStrings = {
+    buy: getLocale('cryptoDotComWidgetBuy'),
+    sell: getLocale('cryptoDotComWidgetSell')
+  }
+
+  enum Percentages {
+    twentyFive = 25,
+    fifty = 50,
+    seventyFive = 75,
+    onehundred = 100
+  }
+
+  const confirmDelay = 15
+
+  const [tradeMode, setTradeMode] = React.useState(TradeModes.BUY)
+  const [tradePercentage, setTradePercentage] = React.useState<number | null>(null)
+  const [tradeAmount, setTradeAmount] = React.useState('')
+  const [showConfirmScreen, setConfirmScreen] = React.useState(false)
+  const [counter, setCounter] = React.useState(confirmDelay)
+
+  let intervalId: any;
+  let timerId: any;
```

Comment:
This is very problematic and will result in bugs, including possibly timers not getting cleared. Every time this function runs this variable is re-initialised to `undefined` (which happens many times, every time any prop value changes or redux action runs). It is set within a `React.useEffect` function (which is good, but that reference is to the variable instance where the `React.useEffect` function was first created (it is not re-created every single time this render function is run, only initially and when a dependency changes). However, the `clearTimers` function is a simple function re-created every time the component is re-rendered and refers to the instance variable in that re-render, and so will not clear the actual timer. These variables should instead use `useRef` which will maintain the value over subsequent renders of this function. This exact example is covered by the React docs here: https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables